alexandria/core/views.py

```diff
@@ -100,6 +108,44 @@ def update(self, request, *args, **kwargs):
 
         return response
 
+    @action(methods=["get"], detail=True)
+    def convert(self, request, pk=None):
+        if not settings.ALEXANDRIA_ENABLE_PDF_CONVERSION:
+            raise ValidationError(_("PDF conversion is not enabled."))
+
+        document = self.get_object()
+        file = document.get_latest_original()
+
+        response = requests.post(
+            settings.ALEXANDRIA_DMS_URL + "/convert",
+            data={"target_format": "pdf"},
+            headers={"authorization": get_authorization_header(request)},
+            files={"file": file.content},
+        )
+
+        if response.status_code == HTTP_401_UNAUTHORIZED:
+            raise AuthenticationFailed(_("Token has expired."))
+
+        response.raise_for_status()
+
+        converted_document = models.Document.objects.create(
+            title={k: v + ".pdf" for k, v in document.title.items()},
+            description=document.description,
+            category=document.category,
+            date=document.date,
+        )
+        file_name = file.name + ".pdf"
+        converted_file = models.File.objects.create(
+            document=converted_document,
+            name=file_name,
+            content=ContentFile(response.content, file_name),
+            mime_type="application/pdf",
+            size=len(response.content),
+        )
+        converted_file.create_thumbnail()
+
+        return Response(status=HTTP_200_OK)
```

Comment:
Would it make sense to return the newly created document? Also, status should be `HTTP_201_CREATED`

Reply:
Returning 201 with the new pdf

Reply:
Ok returning serialized json